src/metrics/encoding/protobuf/aggregated_decoder.go

```diff
@@ -62,6 +62,11 @@ func (d AggregatedDecoder) Value() float64 {
 	return d.pb.Metric.TimedMetric.Value
 }
 
+// Annotation returns the decoded annotation.
+func (d *AggregatedDecoder) Annotation() []byte {
+	return d.pb.Metric.TimedMetric.Annotation
```

Comment:
You'll want to update `ReuseAggregatedMetricProto` and everywhere inside there to properly reset the protobuf for reuse.

That is, everywhere you see `pb.Id = pb.Id[:0]` in `protobuf/reset.go` there should also be a `pb.Annotation = pb.Annotation[:0]` so that there can be zero alloc reuse of the annotation.